File: client/modules/native-todo-form-session/android/src/main/java/expo/modules/nativetodoformsession/NativeTodoFormSessionModule.kt
package expo.modules.nativetodoformsession

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class NativeTodoFormSessionModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("NativeTodoFormSession")

    View(NativeTodoFormSessionView::class) {
      Prop("detailPlaceholderText") { view: NativeTodoFormSessionView, text: String? ->
        view.updateDetailPlaceholderText(text)
      }

      Events(
        "onDismiss",
        "onStateSettled"
      )
    }
  }
}

File: client/modules/native-todo-form-session/ios/NativeTodoFormSessionModule.swift
import ExpoModulesCore

public final class NativeTodoFormSessionModule: Module {
  public func definition() -> ModuleDefinition {
    Name("NativeTodoFormSession")

    View(NativeTodoFormSessionView.self) {
      Prop("detailPlaceholderText") { (view: NativeTodoFormSessionView, text: String?) in
        view.updateDetailPlaceholderText(text)
      }

      Events(
        "onDismiss",
        "onStateSettled"
      )
    }
  }
}
